### PR TITLE
SONARCONF-2 Create GitHub release workflow and Cirrus-ci pipeline

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+qa_plugin_task:
+  <<: *QA_TASK_FILTER
+  <<: *LINUX_2_CPU_4G
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  submodules_script:
+    - git submodule update --init
+  plugin_script:
+    - source cirrus-env QA
```
Comment (nils-werner-sonarsource) [inline .cirrus.yml:74]: We should add the runtime matrix again to test against the latest released version and dogfood.

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    - source set_maven_build_version $BUILD_NUMBER
+    - cd its
+    - mvn verify -P its -X
```
Comment (nils-werner-sonarsource) [inline .cirrus.yml:77]:
```suggestion
    - mvn verify -Pits -Drevision=${PROJECT_VERSION} -Dsonar.runtimeVersion=${SQ_VERSION} -B -e -V
```

File: .cirrus.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  build_script:
+    - source cirrus-env BUILD
+    - regular_mvn_build_deploy_analyze
+  cleanup_before_cache_script: cleanup_maven_repository
+
+qa_plugin_task:
+  <<: *QA_TASK_FILTER
+  <<: *LINUX_2_CPU_4G
+  maven_cache:
```
Comment (nils-werner-sonarsource) [inline .cirrus.yml:69]:
```suggestion
  env:
    matrix:
      - SQ_VERSION: LATEST_RELEASE[8.9]
      - SQ_VERSION: DOGFOOD
  maven_cache:
```

File: .cirrus.yml (hunk 2 above)
Comment (christophe-zurn-sonarsource) [reply .cirrus.yml:77]: I agree we should readd the matrix, but what is this `-Drevision=${PROJECT_VERSION}` paramete used for ? We're not using it anywhere, I don't understand what's its purpose

**Review by christophe-zurn-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**